gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -575,14 +576,27 @@ protected void cleanup(Context context) throws IOException, InterruptedException
      * Run the given list of {@link WorkUnit}s sequentially. If any work unit/task fails,
      * an {@link java.io.IOException} is thrown so the mapper is failed and retried.
      */
-    private void runWorkUnits(List<WorkUnit> workUnits) throws IOException, InterruptedException {
+    private void runWorkUnits(List<WorkUnit> workUnits, Context context) throws IOException, InterruptedException {
       if (workUnits.isEmpty()) {
         LOG.warn("No work units to run");
         return;
       }
 
       String jobId = workUnits.get(0).getProp(ConfigurationKeys.JOB_ID_KEY);
+
+      // Setup and start metrics reporting
+      Properties metricReportingProperties = new WorkUnitState(workUnits.get(0)).getProperties();
```

Comment:
I tried, but it was not getting all the properties. I don't quite understand why (I spend a long time trying to figure it out).